docs/10-lifecycle.md

```diff
@@ -0,0 +1,81 @@
+# Lifecycle
+
+ContainerPilot's core functionality is centered around the concepts of jobs, the events that trigger those jobs or that jobs emit, and watches for the external service discovery catalog (Consul).
+
+
+## What is a job?
+
+A ContainerPilot job is a user-defined process and rules for when to execute it, how to health check it, and how to advertise it to Consul. The rules are intended to allow for flexibility to cover nearly any type of process one might want to run. Some possible job configurations include:
```

Comment:
I have two problems with using the term "alert" here:
- It's backwards from how we're using it. The `job`'s service definition is what registers that the service exists. If anything could be described as an "alert" it's the `watch`.
- "Alert" is a term of art in operations that I don't want to overload.

How about "register" in Consul? This matches up with other projects as well as Consul's own APIs.